src/pybind/nnet3/nnet_nnet_pybind_test.py

```diff
@@ -3,38 +3,155 @@
 # Copyright 2020 JD AI, Beijing, China (author: Lu Fan)
 # Apache 2.0
 
+import kaldi
+import unittest
+from torch.utils.dlpack import to_dlpack
+from torch.utils.dlpack import from_dlpack
 import os
 import sys
 sys.path.insert(0, os.path.join(os.path.dirname(__file__), os.pardir))
 
-import unittest
+try:
+    import torch
+except ImportError:
+    print('This test needs PyTorch.')
+    print('Please install PyTorch first.')
+    print('PyTorch 1.3.0dev20191006 has been tested and is known to work.')
+    sys.exit(0)
+
+
+"""
+input dim=40 name=input
+
+# please note that it is important to have input layer with the name=input
+# as the layer immediately preceding the fixed-affine-layer to enable
+# the use of short notation for the descriptor
+fixed-affine-layer name=lda input=Append(-1,0,1) affine-transform-file=$dir/configs/lda.mat
+
+# the first splicing is moved before the lda layer, so no splicing here
+relu-batchnorm-dropout-layer name=tdnn1 $affine_opts dim=16
+tdnnf-layer name=tdnnf2 $tdnnf_opts dim=16 bottleneck-dim=2 time-stride=1
+linear-component name=prefinal-l dim=4 $linear_opts
+
+prefinal-layer name=prefinal-chain input=prefinal-l $prefinal_opts big-dim=16 small-dim=4
+output-layer name=output include-log-softmax=false dim=$num_targets $output_opts
+
+prefinal-layer name=prefinal-xent input=prefinal-l $prefinal_opts big-dim=16 small-dim=4
+output-layer name=output-xent dim=$num_targets learning-rate-factor=$learning_rate_factor $output_opts
+"""
 
-import kaldi
-from kaldi import read_nnet3_model
-from torch.utils.dlpack import from_dlpack
-from torch.utils.dlpack import to_dlpack
 
 class TestNnetNnet(unittest.TestCase):
 
     def test_nnet_nnet(self):
-        kaldi.SelectGpuId('yes')
-        final_mdl = "/mnt/cfs1_alias1/asr/users/fanlu/task/kaldi_recipe/pybind/s10.1/exp/chain_cleaned_1c/tdnn1c_sp/final.mdl"
+        if torch.cuda.is_available() == False:
+            print('No GPU detected! Skip it')
+            return
+
+        if kaldi.CudaCompiled() == False:
+            print('Kaldi is not compiled with CUDA! Skip it')
+            return
+
+        device_id = 0
+
+        # Kaldi and PyTorch will use the same GPU
+        kaldi.SelectGpuDevice(device_id=device_id)
+        kaldi.CuDeviceAllowMultithreading()
+
+        final_mdl = 'src/pybind/nnet3/final.mdl'
```

Comment:
please change it to `final.mdl`

This test is going to be run inside `pybind/nnet3` directory.

Append a line to `pybind/nnet3/Makefile`.